File: dsa-code-runner/Dockerfile.java
# Use an official lightweight OpenJDK image as the parent image
# We're using Java 17, a modern Long-Term Support (LTS) version
FROM openjdk:17-slim

# Set the working directory inside the container
WORKDIR /usr/src/app

# Default command (will be overridden by our script)
CMD ["/bin/bash"]

File: dsa-code-runner/Dockerfile.cpp
# Use a lightweight base image like Debian Slim
FROM debian:bullseye-slim

# Install the C++ compiler and build essentials
# -y automatically confirms the installation
RUN apt-get update && apt-get install -y g++ build-essential

# Set the working directory inside the container
WORKDIR /usr/src/app

# Default command (will be overridden)
CMD ["/bin/bash"]
